js/sites.js

```diff
@@ -203,11 +203,42 @@
       let enhanced = this.link;
       let url = this.url;
 
+      var getRegularThumbnail = async(function*() {
+        var sw = (yield navigator.serviceWorker.ready).active;
+        var host = new URL(url).host;
+        var thumbURL = new URL(`/pagethumbs/${host}`, window.location);
+
+        var id = Math.random();
+        // Send a message to the SW to check if we have stored this thumb already
+        var thumbInCache = yield new Promise((resolve) => {
+          navigator.serviceWorker.addEventListener("message", function msgHandler({data}) {
+            if (data.name === "SW:HasThumb" && data.id === id) {
+              navigator.serviceWorker.removeEventListener("message", msgHandler);
+              resolve(data.result);
+            }
+          });
+          sw.postMessage({
+            name: "NewTab:HasThumb",
+            id,
+            thumbURL: thumbURL.href,
+          });
+        });
+        if (thumbInCache) {
+          this.showRegularThumbnail(thumbURL);
+          return;
+        }
+        // Capture the page if the thumbnail is missing, which will cause page.js
+        // to be notified and call our refreshThumbnail() method.
+        this.captureIfMissing();
+      }, this);
+
       let title = this.title;
-      if (enhanced && enhanced.title && this._type === ENHANCED) {
+      if (enhanced && enhanced.title && (this._type === ENHANCED || this.link.imageURI)) {
```

Comment:
since enhanced is set to this.link, this can be effectively summarized as:

if ((this._type === ENHANCED || this.link.enhancedImageURI) && this.link.title) { ... }

I think enhancedImageURI is the property to use in case there's an "enhanced tile".